src/nvim/tui/tui.c

```diff
@@ -202,18 +126,94 @@ static void tui_stop(UI *ui)
   }
   xfree(data->write_loop);
   unibi_destroy(data->ut);
-  destroy_screen(data);
+  ugrid_free(&data->grid);
+  data->stop = true;
```

Comment:
Not sure I follow, what does `fg` mean in this context?' When `tui_stop` is called by the ui thread, the call stack should look like this:

- `tui_main`
- `loop_poll_events`
- ...
- `tui_stop`


so when `loop_poll_events` returns the while loop in `tui_main` will exit